src/app/containers/App/App.jsx

```diff
@@ -1,95 +1,65 @@
-/* eslint-disable */
-/*
- * © Jordan Tart https://github.com/jtart
- * https://github.com/jtart/react-universal-app
- */
-import { Component } from 'react';
+import { useEffect, useState, useRef } from 'react';
 import { renderRoutes } from 'react-router-config';
 import { withRouter } from 'react-router-dom';
 import getRouteProps from '../../routes/getInitialData/utils/getRouteProps';
 
-import loadInitialData from '../../routes/loadInitialData';
+export const App = ({ routes, location, initialData, bbcOrigin }) => {
+  const { service, isAmp, id } = getRouteProps(routes, location.pathname);
 
-export class App extends Component {
-  constructor(props) {
-    super(props);
+  const [state, setState] = useState({
+    data: initialData,
+    service,
+    id,
+    isAmp,
+    loading: false,
+    error: null,
+  });
 
-    const { service, isAmp, id } = getRouteProps(
-      this.props.routes,
-      this.props.location.pathname,
-    );
+  const isInitialMount = useRef(true);
 
-    this.state = {
-      data: this.props.initialData,
-      service: service,
-      id: id,
-      isAmp: isAmp,
-      loading: false,
-      error: null,
-      loadInitialDataPromise: null,
-    };
-  }
+  useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
```

Comment:
Rather than using isInitialMount, can you use ref on location pathname instead, and only run if its changed? Something like usePrevious would be nicer IMO
```
const usePrevious = value => {
  const ref = useRef();
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
}
```

Reply:
Source - https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

Reply:
I’m not if this is still needed the initialMount ref is only to ensure we don’t refetch the data on initial component mount. `location.pathname` is set as a dependency of the useEffect hook so that only runs when it changes, except on initial component mount.

Reply:
Its not needed, but usePrevious is a better way to do it IMO

Reply:
Could you add a suggestion of how you would implement this? I'm probably not understanding the pattern as we don't need to return previous values.

Reply:
Looking at it, changed my mind as usePrevious returns undefined until after the first mount. Still dislike `isInitialMount` but not enough to request changed